agent/internal/client/pod_methods.go

```diff
@@ -0,0 +1,317 @@
+	return errors.New("TODO: implement")
+}
+
+// EndTask marks the task as finished with the given status
+func (c *podCommunicator) EndTask(ctx context.Context, detail *apimodels.TaskEndDetail, taskData TaskData) (*apimodels.EndTaskResponse, error) {
+	return nil, errors.New("TODO: implement")
+}
+
+// GetTask returns the active task.
+func (c *podCommunicator) GetTask(ctx context.Context, taskData TaskData) (*task.Task, error) {
+	return nil, errors.New("TODO: implement")
+}
+
+// GetDisplayTaskInfoFromExecution returns the display task info associated
+// with the execution task.
+func (c *podCommunicator) GetDisplayTaskInfoFromExecution(ctx context.Context, td TaskData) (*apimodels.DisplayTaskInfo, error) {
+	return nil, errors.New("TODO: implement")
+}
+
+// GetProjectRef loads the task's project.
+func (c *podCommunicator) GetProjectRef(ctx context.Context, taskData TaskData) (*model.ProjectRef, error) {
+	return nil, errors.New("TODO: implement")
+}
+
+func (c *podCommunicator) GetDistroView(ctx context.Context, taskData TaskData) (*apimodels.DistroView, error) {
+	return nil, errors.New("TODO: implement")
+}
+
+// GetDistroAMI returns the distro for the task.
+func (c *podCommunicator) GetDistroAMI(ctx context.Context, distro, region string, taskData TaskData) (string, error) {
+	return "", errors.New("TODO: implement")
+}
+
+func (c *podCommunicator) GetProject(ctx context.Context, taskData TaskData) (*model.Project, error) {
+	return nil, errors.New("TODO: implement")
+}
+
+func (c *podCommunicator) GetExpansions(ctx context.Context, taskData TaskData) (util.Expansions, error) {
+	return nil, errors.New("TODO: implement")
+}
+
+// Heartbeat sends a heartbeat to the API server. The server can respond with
+// an "abort" response. This function returns true if the agent should abort.
+func (c *podCommunicator) Heartbeat(ctx context.Context, taskData TaskData) (bool, error) {
+	return false, errors.New("TODO: implement")
+}
+
+// FetchExpansionVars loads expansions for a communicator's task from the API server.
+func (c *podCommunicator) FetchExpansionVars(ctx context.Context, taskData TaskData) (*apimodels.ExpansionVars, error) {
+	return nil, errors.New("TODO: implement")
+}
+
+// GetNextTask returns a next task response by getting the next task for a given host.
+func (c *podCommunicator) GetNextTask(ctx context.Context, details *apimodels.GetNextTaskDetails) (*apimodels.NextTaskResponse, error) {
+	return nil, errors.New("TODO: implement")
+}
+
+// GetCedarConfig returns the cedar service information including the base URL,
+// URL, RPC port, and credentials.
+func (c *podCommunicator) GetCedarConfig(ctx context.Context) (*apimodels.CedarConfig, error) {
+	return nil, errors.New("TODO: implement")
+}
+
+// GetCedarGRPCConn returns the client connection to cedar if it exists, or
+// creates it if it doesn't exist.
+func (c *podCommunicator) GetCedarGRPCConn(ctx context.Context) (*grpc.ClientConn, error) {
```

Comment:
Can't this and `createCedarGRPCConn` be implemented in the common communicator?

Reply:
It can't because it depends on `GetCedarConfig`, which can't be implemented by the base communicator because the host one will hit the legacy REST v1 routes (`/api`) while the pod one will hit the newer API (`/rest/v2`).